src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -254,7 +266,8 @@ private Result executeCmd(String cmd) throws IOException, InterruptedException {
                     .stderr(stderr)
                     .cmdAsSingleString(cmd)
                     .start().join() == 0;
-            String stdoutStr = stdout.toString();
+            String stdoutStr = baos.toString(Computer.currentComputer().getDefaultCharset().name());
```

Comment:
Major: <code>UnsupportedEncodingException</code> may happen
Minor: Missing null checks, which may make the static analysis unhappy (I see no real risk).